sympy/stats/tests/test_continuous_rv.py

```diff
@@ -147,12 +151,76 @@ def test_characteristic_function():
     Y = Normal('y', 1, 1)
     cf = characteristic_function(Y)
     assert cf(0) == 1
-    assert simplify(cf(1)) == exp(I - S(1)/2)
+    assert cf(1) == exp(I - S(1)/2)
 
     Z = Exponential('z', 5)
     cf = characteristic_function(Z)
     assert cf(0) == 1
-    assert simplify(cf(1)) == S(25)/26 + 5*I/26
+    assert cf(1).expand() == S(25)/26 + 5*I/26
+
+def test_moment_generating_function():
+    t = S('t')
```

Comment:
This should be declared with appropriate assumptions. As far as I know, `t` should be real, so apply that assumption.

Reply:
I agree with @czgdp1807. Also, I don't think that tests such as these, check if the `mgf` generated is correct or not. perhaps, you can check for actual expressions on Wikipedia or Wolfram and check if the function is generating them.